FutbolRoster/app/src/main/java/shuvalov/nikita/futbolroster/MainActivity.java

```diff
@@ -0,0 +1,122 @@
+import java.util.HashMap;
+
+
+public class MainActivity extends AppCompatActivity {
+    Button addButt;
+    Button remButt;
+    EditText userInp;
+    ListView listView;
+    Spinner spinner;
+
+    BaseAdapter baseAdapter;
+
+    ArrayList<Player> players;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        addButt = (Button) findViewById(R.id.addButt);
+        remButt = (Button) findViewById(R.id.removeButt);
+        userInp = (EditText) findViewById(R.id.userInput);
+        spinner =(Spinner) findViewById(R.id.position);
+
+        listView = (ListView)findViewById(R.id.listView);
+
+        players = new ArrayList<>();
+
+        baseAdapter = new BaseAdapter() {
+            @Override
+            public int getCount() {
+                return players.size();
+            }
+
+            @Override
+            public Object getItem(int position) {
+                return players.get(position);
+            }
+
+            @Override
+            public long getItemId(int position) {
+                return position;
+            }
+
+            @Override
+            public View getView(int position, View convertView, ViewGroup parent) {
+                if(convertView== null){
+                    LayoutInflater layoutInflater = LayoutInflater.from(MainActivity.this);
+                    convertView= layoutInflater.inflate(R.layout.roster_format, null);
+                    }
+                TextView textView1 = (TextView) convertView.findViewById(R.id.pName);
+                TextView textView2 = (TextView) convertView.findViewById(R.id.pPosition);
+                textView1.setText(players.get(position).getName());
+                textView2.setText(players.get(position).getPosition());
+                return convertView;
+                }
+
+            };
+
+        listView.setAdapter(baseAdapter);
+
+        addButt.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                boolean notDuplicate=true;
+                HashMap<String, String> tempPlayInfo = new HashMap<String, String>();
+                for (int i = 0; i<players.size();i++){
+                    tempPlayInfo.put(players.get(i).getName().toLowerCase(), players.get(i).getPosition());
+                    if(tempPlayInfo.containsKey(userInp.getText().toString().toLowerCase()) && tempPlayInfo.get(players.get(i).getName().toLowerCase()).equals(spinner.getSelectedItem().toString())){//If the entered player name exists in the temporary list and that player has the same position assigned to that name, then it's treated as a duplicate. Also added toLowerCase to make case insensitive.
+                        Toast.makeText(MainActivity.this, players.get(i).getName()+" is already on the roster.", Toast.LENGTH_SHORT).show();
+                        notDuplicate=false;
```

Comment:
I'm being picky here, but I think making the variable `boolean duplicate` and setting `duplicate = true` here would be more readable. Obviously your way works too, just takes a minute to think about when reading it.